### PR TITLE
spark/keras: add missing arg shuffle for barekeras dataset

### PR DESCRIPTION
**Pull request by chongxiaoc**
Signed-off-by: Chongxiao Cao <chongxiaoc@uber.com>

## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by chongxiaoc** [comment]
@tgaddair @irasit 
Maybe I'm not familiar with pythonic usage of function handle. But I'm seeing `shuffle` is not passed down when making bare Keras dataset.

**Comment by tgaddair** [comment]
@chongxiaoc are you testing this with standalone keras (not `tensorflow.keras`)? This code path is only used with the standalone keras package.

**Review by tgaddair — APPROVED** [review]
